### PR TITLE
CLN: changed .format to f-string in pandas/core/dtypes

### PR DESCRIPTION
**Pull request by DorAmram**
- [ ] ref #29547
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@DorAmram Thanks for the PR. There are lint failures. Can you run `black pandas`, see https://dev.pandas.io/docs/development/contributing.html#python-pep8-black

File: pandas/core/dtypes/common.py [hunk 1]
```diff
@@ -1859,7 +1859,7 @@ def _validate_date_like_dtype(dtype) -> None:
     try:
         typ = np.datetime_data(dtype)[0]
     except ValueError as e:
-        raise TypeError("{error}".format(error=e))
+        raise TypeError(f"{e}")
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/common.py]: hmm, I not keen on f-strings used like this where equivalent to just `str(e)`. However, in this case, I _think_ you could just pass the ValueError directly to the TypeError constructor.
```suggestion
        raise TypeError(e)
```

Comment (DorAmram) [reply pandas/core/dtypes/common.py]: Will do
Thanks for the comment!

**Comment by jbrockmendel** [comment]
Still needs `black pandas/core/dtypes/common.py` to get the linter to be happy

**Review by simonjayhawkins — APPROVED** [review]
@DorAmram Thanks for the update. lgtm.

**Comment by topper-123** [comment]
Thanks @DorAmram!

**Review by topper-123 — APPROVED** [review]
LGTM.

**Pull request merged**